airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/DefaultAirbyteMapper.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.workers.protocols.airbyte;
+
+import com.google.common.base.Preconditions;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.StandardSyncInput;
+import io.airbyte.config.StandardTargetConfig;
+import io.airbyte.protocol.models.AirbyteCatalog;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteStream;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.workers.WorkerException;
+import java.io.IOException;
+
+/**
+ * We apply some transformations on the fly on the catalog (same should be done on records too) from
+ * the source before it reaches the destination. One of the transformation is to define the
+ * destination namespace where data will be stored and how to mirror (or not) the namespace used in
+ * the source (if any). This is configured in the UI through the syncInput.
+ */
+public class DefaultAirbyteMapper implements AirbyteMapper {
+
+  private StandardSyncInput syncInput;
+
+  public DefaultAirbyteMapper() {}
+
+  @Override
+  public void start(StandardTargetConfig targetConfig, StandardSyncInput syncInput) throws IOException, WorkerException {
+    this.syncInput = syncInput;
+    targetConfig.setCatalog(transformCatalog(targetConfig.getCatalog(), syncInput));
```

Comment:
This is a very bad practice to do inplace transformation.

Maybe the mapper needs 2 methods:
- one to transform the catalog
- one to transform the message